### PR TITLE
Improve Snippets - Second pass

### PR DESCRIPTION
**Pull request by Cocographique**
**Task** [2191381](https://www.odoo.com/web#id=2191381&action=333&active_id=974&model=project.task&view_type=form&cids=1&menu_id=4720)

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by Cocographique** [comment]
Hi @qsm-odoo 
Can you have a look to this and give me your feedback/review?
Thanks.

File: addons/website_sale/views/snippets.xml [hunk 1]
```diff
@@ -6,18 +6,15 @@
...
         <div class="container">
             <div class="row">
                 <div class="col-lg-8 offset-lg-2">
-                    <div class="card">
-                        <div class="card-body">
-                            <h2>Search for a product</h2>
-                            <p>We have amazing products in our shop, check them now !</p>
-                            <t t-call="website_sale.s_products_searchbar_input"/>
-                        </div>
-                    </div>
```
Comment (qsm-odoo) [inline addons/website_sale/views/snippets.xml]: @Cocographique Could you create a commit in the first stable branch where the bug occurs for this part? You can leave this one here, it will go away with forward-port. And remove the uppercase letter in the commit message (in your new PR, not here).

File: addons/web_editor/static/lib/summernote/src/js/Renderer.js [hunk 2]
```diff
@@ -462,6 +462,21 @@ define([
           event: 'imageShape',
           value: 'img-thumbnail'
         });
+        var leafButton = tplIconButton(options.iconPrefix + options.icons.image.shapeLeaf, {
+          title: lang.image.shapeLeaf,
+          event: 'imageShape',
+          value: 'rounded-leaf'
+        });
+        var pillButton = tplIconButton(options.iconPrefix + options.icons.image.shapePill, {
+          title: lang.image.shapeLeaf,
```
Comment (qsm-odoo) [inline addons/web_editor/static/lib/summernote/src/js/Renderer.js]: @Cocographique I'd prefer not to modify the editor bar right now. Also here you use the wrong variable. Could you take this commit out in a new PR? Again: no need to remove it from this one, I will push forced anyway.

File: addons/website/data/website_data.xml [hunk 3]
```diff
@@ -601,6 +601,24 @@
             <field name="type">url</field>
             <field name="url">/website/static/src/img/snippets_demo/s_text_image.jpg</field>
         </record>
+        <record id="website.s_three_columns_image_default_1" model="ir.attachment">
```
Comment (qsm-odoo) [inline addons/website/data/website_data.xml]: @Cocographique you named them image_default here but used default_image in the template so this does not work :/ Anyways, I fixed it.

File: addons/website/data/website_data.xml [hunk 4]
```diff
@@ -601,21 +601,21 @@
...
-            <field name="name">s_three_columns_image_default_2.jpg</field>
+            <field name="name">s_three_columns_default_image_2.jpg</field>
             <field name="type">url</field>
             <field name="url">/web/image/website.library_image_13</field>
         </record>
-        <record id="website.s_three_columns_image_default_3" model="ir.attachment">
+        <record id="website.s_three_columns_default_image_3" model="ir.attachment">
             <field name="public" eval="True"/>
-            <field name="name">s_three_columns_image_default_3.jpg</field>
+            <field name="name">s_three_columns_default_image_3.jpg</field>
```
Comment (qsm-odoo) [inline addons/website/data/website_data.xml]: @Cocographique Oh ok, you renamed to the right name you used in the other commit here ^^"
Anyways, I am not sure this is a good idea to rename those, they are used by people in their template, so you break all website using those for a convention reason :/ I'll leave the commit out my review, you can move it to another PR to be judged later.

File: addons/website/views/snippets/s_image_text.xml [hunk 5]
```diff
@@ -12,7 +12,7 @@
                     <h2>Section Subtitle</h2>
                     <p>Write one or two paragraphs describing your product or services. <br/>To be successful your content needs to be useful to your readers.</p>
                     <p>Start with the customer – find out what they want and give it to them.</p>
-                    <p><a href="#" class="btn btn-primary">Discover more</a></p>
```
Comment (qsm-odoo) [inline addons/website/views/snippets/s_image_text.xml]: @Cocographique No, snippets should be able to use btn-primary. What needs to be fixed is the link dialog. I'll leave this out of the PR :+1:

File: addons/website/views/snippets/s_media_list.xml [hunk 6]
```diff
@@ -37,7 +37,7 @@
                         <i class="fa fa-1x fa-fw fa-user"/>by John DOE
                     </span>
                     <p>Use this component for creating a list of featured elements to which you want to bring attention.</p>
-                    <p><a href="#">Continue reading <i class="fa fa-long-arrow-right align-middle ml-1"/></a></p>
+                    <p><a href="#" class="btn btn-link">Continue reading <i class="fa fa-long-arrow-right align-middle ml-1"/></a></p>
```
Comment (qsm-odoo) [inline addons/website/views/snippets/s_media_list.xml]: The style is kinda broken using btn btn-link on this link. I'll leave that out too. @Cocographique

File: addons/website/views/snippets/s_banner.xml [hunk 7]
```diff
@@ -10,7 +10,7 @@
                     <h1><font style="font-size: 62px;">Sell Online. Easily.</font></h1>
                     <p class="lead">This is a simple hero unit, a simple jumbotron-style component <br/>for calling extra attention to featured content or information.</p>
                     <p class="lead"><br/></p>
-                    <a href="#" class="btn btn-epsilon rounded-circle">Do something</a>
+                    <a href="#" class="btn btn-alpha">Do something</a>
```
Comment (qsm-odoo) [inline addons/website/views/snippets/s_banner.xml]: I don't think the commit message of this is the reason but I'll keep that change @Cocographique

File: addons/website/views/snippets/s_banner.xml [hunk 8]
```diff
@@ -2,19 +2,41 @@
...
 
 <template id="s_banner" name="Banner">
-    <section class="s_banner parallax s_parallax_is_fixed pt96 pb96" data-scroll-background-ratio="1">
+    <section class="s_banner parallax s_parallax_is_fixed o_half_screen_height" data-scroll-background-ratio="1">
         <span class="s_parallax_bg oe_img_bg" style="background-image: url('/web/image/website.s_banner_default_image'); background-position: 50% 0;"/>
         <div class="container">
-            <div class="row s_nb_column_fixed">
-                <div class="col-lg-7 bg-white jumbotron rounded pt32 pb32" data-name="Box">
+            <div class="row no-gutters s_nb_column_fixed">
+                <div class="s_banner_box col-lg-7 bg-white p-5 lead" data-name="Box">
```
Comment (qsm-odoo) [inline addons/website/views/snippets/s_banner.xml]: @Cocographique Can you move this one in another PR, there needs to be lot more discussion about this. Using no-gutters is not a good solution to me. And if it is, it should probably a generic solution for all snippets, no just banner.

File: addons/website/views/snippets/s_cover.xml [hunk 9]
```diff
@@ -2,17 +2,32 @@
...
+                <p>
+                    <a href="/aboutus" class="btn btn-alpha">About us</a>
+                    <a href="/contactus" class="btn btn-beta">Contact us</a>
+                </p>
+            </div>
         </div>
     </section>
 </template>
 
+<template id="s_banner_options" inherit_id="website.snippet_options">
```
Comment (qsm-odoo) [inline addons/website/views/snippets/s_cover.xml]: Banner options in cover? + same comment as banner, put this in another PR @Cocographique

File: addons/website/static/src/scss/bootstrap_overridden.scss [hunk 10]
```diff
@@ -30,6 +30,8 @@ $font-family-sans-serif: $o-theme-font !default;
 $headings-font-family: $o-theme-headings-font !default;
 $headings-color: palette-color('h1') !default;
 
+$lead-font-size: 1.125rem;
```
Comment (qsm-odoo) [inline addons/website/static/src/scss/bootstrap_overridden.scss]: Don't understand this commit. Also we recently remove the use of the lead class because FP find it confusing and not possible to add through the editor... -> other PR with more explanation @Cocographique

File: addons/website/views/snippets/snippets.xml [hunk 11]
```diff
@@ -205,7 +205,7 @@
     </div>
 
     <div data-js="CarouselItem"
-         data-selector=".s_carousel .carousel-item">
+         data-selector=".carousel .carousel-item">
```
Comment (qsm-odoo) [inline addons/website/views/snippets/snippets.xml]: Was like that before, was changed to s_carousel because of image gallery. I leave that commit out, move it to another PR and target quote carousel explicitely or exclude gallery snippet @Cocographique

File: addons/website/views/snippets/s_masonry_block.xml [hunk 12]
```diff
@@ -7,7 +7,7 @@
         <div class="container-fluid">
             <div class="row">
                 <div class="col-lg-6 bg-600 oe_img_bg text-center pb224 pt224" data-name="Block"
-                     style="background-image: url(/web/image/website.s_masonry_block_default_image_1);">
```
Comment (qsm-odoo) [inline addons/website/views/snippets/s_masonry_block.xml]: Again, I'll let this out ;)

File: addons/website/static/src/snippets/s_showcase/002.scss [hunk 13]
```diff
@@ -0,0 +1,45 @@
+.s_showcase[data-vcss='002'] {
```
Comment (qsm-odoo) [inline addons/website/static/src/snippets/s_showcase/002.scss]: As discussed I updated to remove all the scss, and replace with appropriate options.js

File: addons/website/static/src/scss/website.scss [hunk 14]
```diff
@@ -426,6 +426,45 @@ table.table_desc tr td {
 // Snippets
 //
 
+// To Check data-vcss
+.s_title {
```
Comment (qsm-odoo) [inline addons/website/static/src/scss/website.scss]: @sbuhl I have said like 5 times at both you and fabien: are you **sure** you want to remove this as it is working fine? ... and here we reintroduce it?

cc @sdegueldre :smile:

Comment (sbuhl) [reply addons/website/static/src/scss/website.scss]: 😳 I'm sorry 
"faire, défaire et refaire..."

Comment (sdegueldre) [reply addons/website/static/src/scss/website.scss]: IMO this should not be done with classes but with theme customization. Titles are central to a theme's design and they should be consistent across the board. If you want to give the user control, give them control at the theme level (like we do for font and color)

File: addons/website_sale/views/snippets.xml (hunk 1 above)
Comment (Cocographique) [reply addons/website_sale/views/snippets.xml]: @qsm-odoo Do you want me to create a specific task too?
[PR](https://github.com/odoo/odoo/pull/46829)

File: addons/web_editor/static/lib/summernote/src/js/Renderer.js (hunk 2 above)
Comment (Cocographique) [reply addons/web_editor/static/lib/summernote/src/js/Renderer.js]: [**PR 46848**](https://github.com/odoo/odoo/pull/46848)

Comment (Cocographique) [reply addons/web_editor/static/lib/summernote/src/js/Renderer.js]: **PR** [46848](https://github.com/odoo/odoo/pull/46848)

File: addons/website/static/src/scss/website.scss (hunk 14 above)
Comment (qsm-odoo) [reply addons/website/static/src/scss/website.scss]: Nothing to do with theme design to me. It's not a style for all titles, just a one-time title style you need for a specific page. Anyways, we'll check again when this branch remaining commits are moved at the right location (cc @Cocographique :+1:)

File: addons/website/data/website_data.xml (hunk 4 above)
Comment (Cocographique) [reply addons/website/data/website_data.xml]: **PR** [46829](https://github.com/odoo/odoo/pull/46873)

File: addons/website_sale/views/snippets.xml (hunk 1 above)
Comment (qsm-odoo) [reply addons/website_sale/views/snippets.xml]: Yes create a task (or multiple if meaningful) for the commits I did not merge and add the PR link in there (and remove the spec from the original task and add it in the new task(s)). And move it to RD Dev - Green if finished.

File: addons/website/views/snippets/snippets.xml (hunk 11 above)
Comment (Cocographique) [reply addons/website/views/snippets/snippets.xml]: **PR** [46892](https://github.com/odoo/odoo/pull/46892)

File: addons/website/static/src/scss/bootstrap_overridden.scss (hunk 10 above)
Comment (Cocographique) [reply addons/website/static/src/scss/bootstrap_overridden.scss]: **PR** [46902](https://github.com/odoo/odoo/pull/46902)

File: addons/website/views/snippets/s_media_list.xml (hunk 6 above)
Comment (Cocographique) [reply addons/website/views/snippets/s_media_list.xml]: @qsm-odoo Indeed. And it seems that meanwhile, the spec for this snippet evolved to something more "ambitious".

File: addons/website/views/snippets/s_cover.xml (hunk 9 above)
Comment (Cocographique) [reply addons/website/views/snippets/s_cover.xml]: **PR** [46922](https://github.com/odoo/odoo/pull/46922)

File: addons/website/views/snippets/s_banner.xml (hunk 8 above)
Comment (Cocographique) [reply addons/website/views/snippets/s_banner.xml]: **PR** [46922](https://github.com/odoo/odoo/pull/46922)

File: addons/website/views/snippets/s_media_list.xml (hunk 6 above)
Comment (Cocographique) [reply addons/website/views/snippets/s_media_list.xml]: Moved to a dedicated task: [2209958](https://www.odoo.com/web#id=2209958&action=333&active_id=974&model=project.task&view_type=form&cids=1&menu_id=4720)

**Comment by sdegueldre** [comment]
@Cocographique considering the linked task is marked as complete and merged with #46773 I'm guessing this PR can be closed?

**Comment by Cocographique** [comment]
@sdegueldre Yes, indeed!
